### PR TITLE
[prometheus] set lookbackDelta Prometheus Main option to 2x scrapeInterval

### LINKED ISSUE #4951: [prometheus] Make lookback delta equal to x2 scrape interval for main
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

The lookback delta (`--query. lookback-delta`) in Prometheus equals 5 minutes by default.
If there is no metric on the next scrape, Prometeus will still return the value for it for 5 minutes.

It leads to the evaluation failure of rules (both recording and alerting).

P.S. Only the main Prometheus is affected by the problem.

### Proposed Solution

Consider lowering the delta. As per VictoriaMetrics suggestion, using the 2x scrape interval is sane.

### Additional Information

https://github.com/VictoriaMetrics/VictoriaMetrics/issues/209#issuecomment-542345971

### PR DESCRIPTION
**Pull request by miklezzzz**
## Description
Sets query.lookbackDelta parameter to the value equal to double `Values.global.discovery.prometheusScrapeInterval` value. Prometheus Main restart will take place.
Also, the validation pattern for `scrapeInterval` has been introduced (the same to what Prometheus CRD applies to the corresponding field).

## Why do we need it, and what problem does it solve?
We adjust `loopbackDelta` value from default `5m` to 2x `scrapeInterval` (in common case, to `60s`, as default `scrapeInterval value is `30s`), as it's said that a lower value improves the rule evaluation functionality.
Close #4951 

## What is the expected result?
Prometheus Main statefulset has `--query.lookback-delta` flag set with the value equal to double `scrapeInterval` value (in most cases).

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: prometheus
type: feature
summary: Sets Prometheus Main parameter 'lookbackDelta' value to double the scrapeInterval value.
impact: Prometheus main restart is required.
impact_level: low
```

### CODE REVIEW

**Review by nabokihms — CHANGES_REQUESTED** [review]

File: modules/300-prometheus/hooks/set_lookback_delta.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	var lookBackDelta time.Duration
+
+	scrapeInterval, err := time.ParseDuration(input.Values.Get("prometheus.scrapeInterval").String())
+	if err != nil {
+		return nil
+	}
+	if scrapeInterval < minLookbackDelta {
+		lookBackDelta = minLookbackDelta
+	} else {
+		lookBackDelta = scrapeInterval * 2
```
[the remark this answers is not in the document]
Comment (miklezzzz) [reply modules/300-prometheus/hooks/set_lookback_delta.go:43]: and what if scrapeInterval is set to `5m30s`?

[the remark this answers is not in the document]
Comment (miklezzzz) [reply modules/300-prometheus/hooks/set_lookback_delta.go:43]: and what do you think of idea to have some reasonable minimum lookbackDelta value imposed? while it's possible to set scrapeInterval to something like 10ms, I assumed that having 20ms set for lookbackDelta wouldn't be OK.

[the remark this answers is not in the document]
Comment (nabokihms) [reply modules/300-prometheus/hooks/set_lookback_delta.go:43]: Didn't get the problem with 5m30s. Is this not valid? X2 interval will be 11 minutes.

About defaulting, I asked a question below. Is there a problem with low values? AFAIK it is totally ok, and this is the approach VictoriaMetrics uses.

**Comment by miklezzzz** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.23 5356917480 5356917480-1-con-1-23 /sys/deckhouse-oss/install:pr5006 cloud-user@84.201.159.16

File: modules/300-prometheus/hooks/set_lookback_delta.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (miklezzzz) [reply modules/300-prometheus/hooks/set_lookback_delta.go]: @nabokihms I've updated the PR and its description, ready for review.

**Review by nabokihms — APPROVED** [review]

**Review by yalosev — APPROVED** [review]

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.23 5356917480 5356917480-1-con-1-23 /sys/deckhouse-oss/install:pr5006 cloud-user@84.201.159.16

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Pull request merged**